src/collective/documentgenerator/utils.py

```diff
@@ -66,3 +68,9 @@ def update_templates(templates, profile='', force=False):
             modified(obj)
             ret.append((ppath, ospath, 'replaced'))
     return ret
+
+
+def validate_dict_update(original_dict, update_dict, error_message='Overwrite value is not allowed.'):
```

Comment:
error_message=_("Dict update collision on key")

method name could be "update_dict_with_validation" and update could be done in it. See later